llvm/lib/Transforms/Vectorize/VectorCombine.cpp

```diff
@@ -969,7 +970,11 @@ class ScalarizationResult {
 public:
   ScalarizationResult(const ScalarizationResult &Other) = default;
   ~ScalarizationResult() {
-    assert(!ToFreeze && "freeze() not called with ToFreeze being set");
+    // The object may be copied to another scope if it is in state
+    // StatusTy::SafeWithFreeze.
+    if (Status != StatusTy::SafeWithFreeze)
```

Comment:
When this class object is used in `std::make_pair` at line 1182, `std::make_pair` will create a temp local object in its stackframe, then the assert will be triggered. So I have to avoid this via adding a check before the assert.
To a

Reply:
Can we move the value instead? Maybe using try_emplace instead of insert would work?

Reply:
Using `std::move` and `try_emplace` do work. No need to modify the destructor. Thanks!